wpilibj/src/main/java/edu/wpi/first/wpilibj/event/BooleanEvent.java

```diff
@@ -155,6 +159,21 @@ public BooleanEvent negate() {
    * @param other the event to compose with
    * @return the event that is active when both events are active
    */
+  public BooleanEvent and(BooleanEvent other) {
```

Comment:
This overload won't be resolved correctly if a BooleanEvent is stored as a BooleanSupplier.
See my comment about C++.

Reply:
Not sure what you mean. I really do need that overload because if the BooleanEvents use different loops, the 2nd BooleanEvent's state variable needs to be updated.

Reply:
I meant make the return `m_state.get() && ...` instead of `m_signal.getAsBoolean() && ...`. As for Starlight's comment, overloads (unlike overrides) are picked at compile time, so it would call the `BooleanSupplier` whenever the compile-time type of the object is `BooleanSupplier`, even if its runtime type would actually fit a different type. See https://godbolt.org/z/xxWes65qf.